From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from mail-oa1-f52.google.com (mail-oa1-f52.google.com [209.85.160.52])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 5B5831FAC42
	for <linux-wireless@vger.kernel.org>; Thu, 19 Feb 2026 00:40:14 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=none smtp.client-ip=209.85.160.52
ARC-Seal:i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1771461615; cv=none; b=EoaSxP26KEMOuASPWUQJMhxUfVe9I58j2N2iKW5OWF97WbPVE3DXUtq0tJ+1pqpDAntAT76YTzYJjH2RAH5ODYL2lkEGgx5Vv2QqR61G5kWSOuoYG7mDwIiK9Y8E4PCdlgseu/bu1t0kHzK6qv2fHxATaU2VKdamABzpw8vQgTs=
ARC-Message-Signature:i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1771461615; c=relaxed/simple;
	bh=PVb3kWFSh3+MEQECSeLvEjmh80ilAhH7l158hufcovU=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version; b=Kk/cDZTlZAoKd1ocxkbd4sWax9o8X3WNKz0pAyUCkw3L+1/Dlx3tu9OZ/sy0IUk8EJ+t7HIRvMXtethtQ/ndMRs8e1HlArxhAZ+AD4MQPGTCnWKV+fpYg4hpwcSQEriqcfFmmYD4Xp/GoEqnr+NB68W62VSmqYZpc33TGDQEzRo=
ARC-Authentication-Results:i=1; smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org; spf=pass smtp.mailfrom=gmail.com; arc=none smtp.client-ip=209.85.160.52
Authentication-Results: smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org
Authentication-Results: smtp.subspace.kernel.org; spf=pass smtp.mailfrom=gmail.com
Received: by mail-oa1-f52.google.com with SMTP id 586e51a60fabf-40f387a688dso293630fac.0
        for <linux-wireless@vger.kernel.org>; Wed, 18 Feb 2026 16:40:14 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1771461613; x=1772066413;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-gg:x-gm-message-state:from
         :to:cc:subject:date:message-id:reply-to;
        bh=BqnEDY5rPm6avb1ZLa7TTBT9hEe6l1pmetOvQRaRsnM=;
        b=DgjJLxpoyHILmL8yV0lPKuFVaQouRj0hJ9U2Rug9lTCOORxWjMQZMMtUmSaFrMOGHS
         Ec8/G/xY1Ni6iDDW8b+bGxIsU4bSBHFtb1wXuDDMJZh889PMDSwPBlp8iot8AR6TOdBC
         R1xJjy+40m+5kc3RwW6Qs+RdRp1GpmabqMZ1b3Bja9D6EV1XrtqfXdMY7puLDJs2S03I
         oJWMu9tb7iUTg8qfckKG2+GgCtcBiivBxvCyCM/J2KcOV61xcV/Obo4UvCoO2y2ycWWV
         0p08E4fxbWjzvGr98RfjgEqCfkjhWOkPg+rx4XnhLnLl26vWRu+88LX30APbHn6ghQQW
         s/NA==
X-Gm-Message-State: AOJu0YyKMO6o2lo7jLOzXmIzGVUISIq783BxRRsT7NSwoULFMFHGQwYq
	P6jnpyaEIA7imc7o6J5Dr/ijMsMZsLuEU5MLwqflhdmQlCMlhuTgO1BD
X-Gm-Gg: AZuq6aLBJvMMb3bwy2gfCENHonBMFvSctRHeX0vrSF25zivNkpxpKIGzRAiVMVOUkFu
	Nvqf79vq5D9/qqq58O5YgRT8z6I02DrhIhVq0ZteiybYgGu5V5Wj4YZeoEOP90fhxTBJtBNtzL5
	WkRUVc3oZxV9aMYDUGN9nrNTZZRO7c27b6+HE0ludTSHDb6+pxMSjdJae0CicKd1t4JCH37fZvP
	k4kPNHzNaX+QUQu/5w0xNWmBkj76wUpk8ThbIZtOJlw4+47qBUIv7jDVEnf6FXkXD43sOiq1z4z
	xk+VWe+rdE+ntBBPtGdhjZM0BsMs3nCa8j7ZSP588LPebKAOlZwKP/YLHUZR3rU6DzMJ6csO0f6
	37/kq3ZFlPl9pzf7Ib1ODTHG4RljJ0Nn0yzqDBwFV3vmk4KY5J29gFAGuktP9aQsequVe/AWivO
	AQM125p8YuN3sVe2d1cupvO2sswH6HcJXd6NiQ1kt+zdP4J5Y=
X-Received: by 2002:a05:6871:c941:b0:409:60ad:2b1e with SMTP id 586e51a60fabf-41539d180f1mr622220fac.18.1771461613219;
        Wed, 18 Feb 2026 16:40:13 -0800 (PST)
Received: from sean-HP-EliteBook-830-G6.lan ([207.191.35.252])
        by smtp.gmail.com with ESMTPSA id 586e51a60fabf-40eaee45d12sm23477659fac.1.2026.02.18.16.40.11
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Feb 2026 16:40:12 -0800 (PST)
From: sean.wang@kernel.org
To: nbd@nbd.name,
	lorenzo.bianconi@redhat.com
Cc: linux-wireless@vger.kernel.org,
	linux-mediatek@lists.infradead.org,
	Sean Wang <sean.wang@mediatek.com>
Subject: [PATCH 02/11] wifi: mt76: mt7921: use mt76_for_each_q_rx() in reset path
Date: Wed, 18 Feb 2026 18:39:58 -0600
Message-ID: <20260219004007.19733-2-sean.wang@kernel.org>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20260219004007.19733-1-sean.wang@kernel.org>
References: <20260219004007.19733-1-sean.wang@kernel.org>
Precedence: bulk
X-Mailing-List: linux-wireless@vger.kernel.org
List-Id: <linux-wireless.vger.kernel.org>
List-Subscribe: <mailto:linux-wireless+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-wireless+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

From: Sean Wang <sean.wang@mediatek.com>

Replace explicit napi_disable() calls for RX queues with
mt76_for_each_q_rx() in mt7921e_mac_reset().

This removes hardcoded queue indices and disables all configured
RX queues during reset.

Signed-off-by: Sean Wang <sean.wang@mediatek.com>
---
 drivers/net/wireless/mediatek/mt76/mt7921/pci_mac.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/mediatek/mt76/mt7921/pci_mac.c b/drivers/net/wireless/mediatek/mt76/mt7921/pci_mac.c
index 5ec084432ae3..0db7acb3a637 100644
--- a/drivers/net/wireless/mediatek/mt76/mt7921/pci_mac.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7921/pci_mac.c
@@ -71,9 +71,9 @@ int mt7921e_mac_reset(struct mt792x_dev *dev)
 	mt76_txq_schedule_all(&dev->mphy);
 
 	mt76_worker_disable(&dev->mt76.tx_worker);
-	napi_disable(&dev->mt76.napi[MT_RXQ_MAIN]);
-	napi_disable(&dev->mt76.napi[MT_RXQ_MCU]);
-	napi_disable(&dev->mt76.napi[MT_RXQ_MCU_WA]);
+	mt76_for_each_q_rx(&dev->mt76, i) {
+		napi_disable(&dev->mt76.napi[i]);
+	}
 	napi_disable(&dev->mt76.tx_napi);
 
 	mt76_connac2_tx_token_put(&dev->mt76);
-- 
2.43.0


